### DISCUSSION TITLE
Feature request: add an option for stricter profit switching in mev-geth

### DISCUSSION THREAD

**Discussion by jparyani** [flashbots/pm#56] [miners]
Right now, mev-geth proposes blocks as fast as it can. The profit switcher in mev-geth will eventually converge to the most profitable block, but there can be timing windows where it proposes a less profitable block before the vanilla block is done being calculated. This leads to problems where miners will be working on a suboptimal block for a short amount of time. This is compounded by many miners blocking quick updates, since there's a switching cost to loading a new block.

Proposal: add an option for miners to wait until both the flashbots and vanilla worker have computed the block before proposing the winner. A quick spot check shows this adds around 50ms of latency on average, but it can sometimes go way higher (up to 1+ seconds).

Issue: https://github.com/flashbots/mev-geth/issues/38

**Comment by thegostep** [comment]
# [v0.2 Proposal](https://hackmd.io/@flashbots/core-v2-proposal#Strict-profit-switching)

v0.2 introduces a new geth config `miner.strictprofitswitch` which forces the mev-geth multi-worker to wait for the vanilla block to be completed before comparing the flashbots block.

This eliminates the race condition where a flashbots block could be mined before the vanilla block is done being calculated.
